cli/2024-04-azure-workload-identity.md

```diff
@@ -0,0 +1,233 @@
+
+### Implementation Details
+
+#### UCP
+
+UCP should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+The credential storage and management code will need to be updated to store a ClientID without a ClientSecret.
+
+#### Bicep
+
+n/a
+
+#### Deployment Engine
+
+Deployment Engine should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+#### Core RP
+
+Core RP should be updated to read the Entra ID Application Client ID from the credential store. It should then authenticate with Azure using this Client ID and the signed service account token.
+
+#### Portable Resources / Recipes RP
+
+n/a
+
+### Error Handling
+
+#### AzWI mutating admission webhook is not installed in the cluster.
+* At Radius install time if Radius detects that the AzWI mutating admission webhook is not installed on the cluster, we should return an error to the user: "Azure Workload Identity mutating admission webhook is not installed in the cluster. Please follow the guidance at aka.ms/rad-workload-identity to set up workload identity for Radius."
+
+## Test plan
+
+We should add a functional test that installs Radius with Azure Workload Identity enabled and then deploys an Azure resource to ensure that the resource is deployed successfully. We will need to maintain an Entra ID Application in Azure for testing purposes.
+
+## Security
+
+We will not be storing any secrets in Radius. The user will have to register the Entra ID Application Client ID with Radius using the `rad credential register azure` command. This command will not require a client secret. We are using a project and process that is a well-known way to authenticate with Azure via Azure Workload Identity. The user will have the same security concerns as today - their Entra ID Application should have least-privileges.
+
+## Compatibility
+
+There should be no breaking changes as a result of this design.
+
+To use Azure Workload Identity with Radius, the user will have to re-install the Radius control plane.
+
+## Monitoring and Logging
+
+We will have the same monitoring and logging as today. We will not be adding any new instrumentation.
+
+## Development plan
+
+* Create POC for Radius + Azure Workload Identity (1 engineer, 0.5 sprint)
+* Create and review technical design (1 engineer, 0.5 sprint)
+* Implement CLI and Helm chart changes (1 engineer, 1 sprint)
+* Implement changes in UCP, Bicep, and Applications RP (1 engineer, 0.5 sprint)
+* End-to-end testing and documentation (1 engineer, 0.5 sprint)
+
+Total: 3 sprints of engineering effort
+
```

Comment:
Do we want to describe Test Plan/approach in some detail

Reply:
yes, covered here: https://github.com/radius-project/radius/issues/7715